### PR TITLE
feat: Support issuesless events

### PR DESCRIPTION
**Pull request by dcramer**
some initial draft work to see what it'd take - Snuba currently rejects these events

clickhouse_driver.errors.TypeMismatchError: Code: 53. Type mismatch in VALUES section. Repeat query with types_check=True for detailed info. Column group_id: cannot convert argument to integer
[pid: 22|app: 0|req: 16/16] 172.19.0.1 () {28 vars in 335 bytes} [Thu May  9 14:24:54 2019] POST /tests/eventstream => generated 0 bytes in 55 msecs (HTTP/1.1 500) 0 headers in 0 bytes (0 switches on core 0)

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -752,34 +752,44 @@ def save(self, project_id, raw=False, assume_normalized=False):
...
-            group, is_new, is_regression, is_sample = self._save_aggregate(
-                event=event, hashes=hashes, release=release, **kwargs
-            )
-        except HashDiscarded:
-            event_discarded.send_robust(
-                project=project,
-                sender=EventManager,
-            )
+        should_save = event.get_event_type() != 'transaction'
+        should_save = False
```
Comment (dcramer) [inline src/sentry/event_manager.py]: note this is where its hardcoded to save no groups

File: src/sentry/eventstream/snuba.py [hunk 2]
```diff
@@ -256,6 +256,7 @@ def _send(self, project_id, _type, extra_data=(), asynchronous=True):
                 raise snuba.SnubaError("HTTP %s response from Snuba!" % resp.status)
             return resp
         except urllib3.exceptions.HTTPError as err:
+            raise
```
Comment (dcramer) [inline src/sentry/eventstream/snuba.py]: i added this because the error raised was obscure

File: src/sentry/event_manager.py [hunk 3]
```diff
@@ -907,31 +928,35 @@ def save(self, project_id, raw=False, assume_normalized=False):
...
                 event_id=event.id,
                 tags=event.tags,
                 date_added=event.datetime,
             )
 
         if event_user:
+            metrics_ = [
+                (tsdb.models.users_affected_by_project, project.id, (event_user.tag_value, )),
+            ]
+            if group:
```
Comment (alex-hofsteede) [inline src/sentry/event_manager.py]: These metrics should be gone from redis TSDB entirely I think as they are computed from snuba, You can probably remove these calls altogether.

Comment (dcramer) [reply src/sentry/event_manager.py]: yeah theres some other tagstore stuff in here that i assume also is going to get removed?

**Comment by alex-hofsteede** [comment]
It's probably reasonable to put 0 as the group id if you want snuba to accept these, but not 100% sure that won't cause other issues down the line, there are a lot of references to project.group in API code that will all start raising DoesNotExist

**Comment by dcramer** [comment]
expected, but this errors:

```
16:41:24 worker          |   File "/Users/dcramer/Development/sentry/src/sentry/tasks/base.py", line 54, in _wrapped
16:41:24 worker          |     result = func(*args, **kwargs)
16:41:24 worker          |   File "/Users/dcramer/Development/sentry/src/sentry/tasks/post_process.py", line 123, in post_process_group
16:41:24 worker          |     event.group, _ = get_group_with_redirect(event.group_id)
16:41:24 worker          |   File "/Users/dcramer/Development/sentry/src/sentry/models/group.py", line 77, in get_group_with_redirect
16:41:24 worker          |     raise error  # raise original `DoesNotExist`
16:41:24 worker          | Exception: Group matching query does not exist.
16:41:24 worker          | 16:41:24 [ERROR] celery.worker.job: Task sentry.tasks.post_process.post_process_group[570d461c-1614-4b14-be94-da8841efca7f] raised unexpected: DoesNotExist('Group matching query does not exist.',) (data={u'hostname': 'celery@MacBook-Pro-5.local', u'name': 'sentry.tasks.post_process.post_process_group', u'args': '[]', u'internal': False, u'kwargs': "{'event': <Event at 0x10a169210: id=187L, project_id=1L, group_id=0>, 'is_new_group_environment': None, 'is_new': None, 'primary_hash': 'f9633dad8cc5e1ceeac4325129941d53', 'is_sample': False, 'is_regression': None}", u'id': '570d461c-1614-4b14-be94-da8841efca7f'})
```

File: src/sentry/event_manager.py [hunk 4]
```diff
@@ -860,31 +877,36 @@ def save(self, project_id, raw=False, assume_normalized=False):
...
+            tsdb.record_frequency_multi(frequencies, timestamp=event.datetime)
+
+        if group:
+            UserReport.objects.filter(
+                project=project,
+                event_id=event_id,
+            ).update(
+                group=group,
+                environment=environment,
+            )
```
Comment (markstory) [inline src/sentry/event_manager.py]: If we get a user-report for a group-less event it won't ever end up being displayed. We might want to account for that in how user-feedback displays.

Comment (dcramer) [reply src/sentry/event_manager.py]: Yea - a follow up for sure. I think we need tweak user reports to get marked as valid once they're paired with an event.

**Comment by dcramer** [comment]
@fpacifici is taking this over so closing out this PR

**Pull request closed without merging**